### ISSUE TITLE
[1.3.3 windows 10]Problem with meteor update - npm-node-aes-gcm@0.1.3_6

### ISSUE THREAD

**Issue by tcastelli** [meteor/meteor#7217]
I've run into this problem after updating a project from 1.3.2.4 to 1.3.3.
Have tried to clean local folder and try meteor update again but result is always the same. (My username in windows has spaces not sure if that's the trouble maker). 
I am using oauth-encryption in my package list( it was working just fine previously)

D:\Programming\workspace\project>meteor update
=> Errors while initializing project:

```
While loading package npm-node-aes-gcm@0.1.3_6:
error: Command failed: gyp
npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Users\\xx
xx\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.3.3\\mt-os.windows.x86_32\\dev_bundle\\bin\\\\node.exe"
"C:\\Users\\xx
xx\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.3.3\\mt-os.windows.x86_32\\dev_bundle\\lib\\node_modules\\npm\\bin\\npm-cli.js"
"rebuild"
npm ERR! node v0.10.45
npm ERR! npm  v2.15.1
npm ERR! code ELIFECYCLE
npm ERR! node-aes-gcm@0.1.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-aes-gcm@0.1.3 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the node-aes-gcm package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs node-aes-gcm
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!
npm ERR!     npm owner ls node-aes-gcm
npm ERR! There is likely additional logging output above.

gyp
npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Users\\xx
xx\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.3.3\\mt-os.windows.x86_32\\dev_bundle\\bin\\\\node.exe"
"C:\\Users\\xx
xx\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.3.3\\mt-os.windows.x86_32\\dev_bundle\\lib\\node_modules\\npm\\bin\\npm-cli.js"
"rebuild"
npm ERR! node v0.10.45
npm ERR! npm  v2.15.1
npm ERR! code ELIFECYCLE
npm ERR! node-aes-gcm@0.1.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-aes-gcm@0.1.3 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the node-aes-gcm package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs node-aes-gcm
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!
npm ERR!     npm owner ls node-aes-gcm
npm ERR! There is likely additional logging output above.
```

**Comment by tcastelli** [comment]
Following up, I tried to do meteor npm install node-aes-gcm
And found the following errors(may be linked with the problem since I see something about visual studio that has been mentioned in others issues)

D:\Programming\webstorm_workspace\project\node_modules\node-aes-gcm>if not defined npm_config_node_gyp (node "C:\Users\xx xx\AppData\Local\.meteor\packages\meteor-tool\1.3.2_4\mt-os.windows.x86_32\dev_bundle\lib\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild )
Los proyectos de esta solución se van a compilar de uno en uno. Para habilitar la compilación en paralelo, agregue el modificador "/m".
MSBUILD : error MSB3428: No se pudo cargar el componente "VCBuild.exe" de Visual C++. Para corregir este problema, 1) i
nstale .NET Framework 2.0 SDK, 2) instale Microsoft Visual Studio 2005 o 3) agregue la ubicación del componente a la ru
ta de acceso del sistema si está instalado en otro lugar.  [D:\Programming\webstorm_workspace\spandlo\node_modules\node
-aes-gcm\build\binding.sln]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\xx xx\AppData\Local\.meteor\packages\meteor-tool\1.3.2_4\mt-os.windows.x86_32\dev_bundle\lib\node_modules\npm\node_modules\node-gyp\lib\build.js:276:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:820:12)
gyp ERR! System Windows_NT 10.0.10586
gyp ERR! command "node" "C:\\Users\\xx xx\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.3.2_4\\mt-os.windows.x86_32\\dev_bundle\\lib\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd D:\Programming\webstorm_workspace\spandlo\node_modules\node-aes-gcm
gyp ERR! node -v v0.10.43
gyp ERR! node-gyp -v v3.3.0
gyp ERR! not ok
npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Users\\xx xx\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.3.2_4\\mt-os.windows.x86_32\\dev_bundle\\bin\\\\node.exe" "C:\\Users\\xx xx\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.3.2_4\\mt-os.windows.x86_32\\dev_bundle\\lib\\node_modules\\npm\\bin\\npm-cli.js" "install" "node-aes-gcm"
npm ERR! node v0.10.43
npm ERR! npm  v2.14.22
npm ERR! code ELIFECYCLE

npm ERR! node-aes-gcm@0.1.7 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-aes-gcm@0.1.7 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the node-aes-gcm package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs node-aes-gcm
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!
npm ERR!     npm owner ls node-aes-gcm
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\Programming\webstorm_workspace\spandlo\npm-debug.log

**Comment by benjamn** [comment]
This is probably related to https://github.com/meteor/meteor/issues/7007, if I had to guess.

You may be able get this to succeed in the meantime by making sure the [prerequisites for `node-gyp`](https://github.com/nodejs/node-gyp/blob/master/README.md#installation) are installed on your system.

**Comment by tcastelli** [comment]
Thanks @benjamn I could finally do a meteor update by following your indications and some other extra steps, for anyone having the same problem this is what I did for Windows 10
1) Install [python 2.7](https://www.python.org/ftp/python/2.7.11/python-2.7.11.amd64.msi) an make sure the path is correct  #7178 is a good reference place to check what needs to be set(it is normally set by default)
2) Install the [Visual Studio C++ Build tools](http://go.microsoft.com/fwlink/?LinkId=691126)
3) Install [OpenSSL-Win32 Lib](https://slproweb.com/download/Win32OpenSSL-1_0_2h.exe) (don't change the default installation path and choose to keep your libs under /bin not on windows folder)
4) At this point, to make sure everything will be fine when doing meteor update, I went to my project folder and did meteor npm install -g node-gyp. If everything was installed properly you will see a list of installed modules
5) Now to test for aes, run meteor npm install node-aes-gcm. You may see some yellow warnings but if it finishes it will be okay.
6) meteor update and cross fingers  for a updated to Meteor 1.3.3. message :)

**Comment by benjamn** [comment]
This should be fixed in Meteor 1.3.3.1 (with `oauth-encryption@1.0.12` and `npm-node-aes-gcm@0.1.5_1`), which you can try by running `meteor update --release 1.3.3.1`.
